File: approvalcrest-junit-jupiter/src/test/java/com/github/karsaig/approvalcrest/MetaInfoTest.java
package com.github.karsaig.approvalcrest;

import static com.github.karsaig.approvalcrest.MatcherAssert.assertThat;
import static com.github.karsaig.approvalcrest.matcher.Matchers.sameBeanAs;

import java.nio.file.Paths;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInfo;

public class MetaInfoTest {

    @Test
    public void testJunitJupiterMetaWithSameBeanAsMatcher() {
        JunitJupiterTestMeta underTest = new JunitJupiterTestMeta();

        JunitJupiterTestMeta expected = new JunitJupiterTestMeta(Paths.get("src/test/java/com/github/karsaig/approvalcrest"), "com.github.karsaig.approvalcrest.MetaInfoTest", "testJunitJupiterMetaWithSameBeanAsMatcher", Paths.get("src/test/resources/approvalcrest"));
        assertThat(underTest, sameBeanAs(expected));
    }

    @Test
    public void testJunit5InfoMetaWithSameBeanAsMatcher(TestInfo testInfo) {
        Junit5InfoBasedTestMeta underTest = new Junit5InfoBasedTestMeta(testInfo);

        Junit5InfoBasedTestMeta expected = new Junit5InfoBasedTestMeta(Paths.get("src/test/java/com/github/karsaig/approvalcrest"), "com.github.karsaig.approvalcrest.MetaInfoTest", "testJunit5InfoMetaWithSameBeanAsMatcher", Paths.get("src/test/resources/approvalcrest"));
        assertThat(underTest, sameBeanAs(expected));
    }
}

File: approvalcrest-junit-kotlin/src/test/kotlin/com/github/karsaig/approvalcrest/kotlin/MetaInfoTest.kt
package com.github.karsaig.approvalcrest.kotlin

import com.github.karsaig.approvalcrest.jupiter.MatcherAssert
import com.github.karsaig.approvalcrest.jupiter.matcher.Matchers
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInfo
import java.nio.file.Paths

class MetaInfoTest {
    @Test
    fun testKotlinMetaWithSameBeanAsMatcher() {
        val underTest = KotlinTestMeta()
        val expected = KotlinTestMeta(
            Paths.get("src/test/kotlin/com/github/karsaig/approvalcrest/kotlin"),
            "com.github.karsaig.approvalcrest.kotlin.MetaInfoTest",
            "testKotlinMetaWithSameBeanAsMatcher",
            Paths.get("src/test/resources/approvalcrest")
        )
        MatcherAssert.assertThat(underTest, Matchers.sameBeanAs(expected))
    }

    @Test
    fun testKotlinInfoBasedMetaWithSameBeanAsMatcher(testInfo: TestInfo) {
        val underTest = KotlinInfoBasedTestMeta(testInfo)
        val expected = KotlinInfoBasedTestMeta(
            Paths.get("src/test/kotlin/com/github/karsaig/approvalcrest/kotlin"),
            "com.github.karsaig.approvalcrest.kotlin.MetaInfoTest",
            "testKotlinInfoBasedMetaWithSameBeanAsMatcher",
            Paths.get("src/test/resources/approvalcrest")
        )
        MatcherAssert.assertThat(underTest, Matchers.sameBeanAs(expected))
    }
}